### PR TITLE
updated reward system per request

### PR DESCRIPTION
**Pull request by AnsonCheah**

### CODE REVIEW

**Comment by AnsonCheah** [comment]
LGTM!!!!!

File: .vscode/settings.json
Comment (PiusLim373) [inline .vscode/settings.json:1]: plese dont push the .vscode folder

File: drone_control_gym.py [hunk 1]
```diff
@@ -159,10 +157,33 @@ def get_pose(self):
...
 
     def step(self, action):
+        self.last_drone_motor_thrust = copy.deepcopy(self.drone_motor_thrust)
+        self.last_drone_pose = copy.deepcopy(self.drone_pose)
+        self.last_imu_acc_data = copy.deepcopy(self.imu_acc_data)
+        self.last_imu_gyro_data = copy.deepcopy(self.imu_gyro_data)
+        ic(self.last_drone_motor_thrust)
+        ic(self.last_drone_pose)
+        ic(self.last_imu_acc_data)
+        ic(self.last_imu_gyro_data)
```
Comment (PiusLim373) [inline drone_control_gym.py]: too much logging

File: drone_control_gym.py [hunk 2]
```diff
@@ -171,28 +192,25 @@ def step(self, action):
...
         self.step_count += 1
 
-        self._update_drone_pose_from_sim()
-        self.drone_motor_thrust = list(self.drone.actuator_force)
+        self._update_drone_data_from_sim()
+        self.drone_motor_thrust = np.array(self.drone.actuator_force)
+        ic(self.drone_motor_thrust)
+        ic(self.drone_pose)
+        ic(self.imu_acc_data)
+        ic(self.imu_gyro_data)
```
Comment (PiusLim373) [inline drone_control_gym.py]: too much logging

File: drone_control_gym.py [hunk 3]
```diff
@@ -85,10 +88,12 @@ def __init__(self):
         self.drone_last_ctrl = copy.deepcopy(self.drone.ctrl)
         self.step_count = 1
 
-    def _update_drone_pose_from_sim(self):
-        rpy_angles = np.rad2deg(quaternion_to_rpy(self.drone.xquat[1]))
+    def _update_drone_data_from_sim(self):
+        rpy_angles = quaternion_to_rpy(self.drone.xquat[1])
         self.drone_rpy = np.array(rpy_angles)
         self.drone_pose = np.array(self.drone.xpos[1])
+        self.imu_acc_data = self.drone.sensordata[self.imu_acc_start : self.imu_acc_start + 3]  # Accelerometer (x, y, z)
```
Comment (PiusLim373) [inline drone_control_gym.py]: can we fix the naming convention? eg: self.drone_acc and self.drone_gyro

File: drone_control_gym.py [hunk 4]
```diff
@@ -171,28 +192,25 @@ def step(self, action):
...
-        self.has_finished, self.reward = self._calculate_reward()  # return bool and float
-        
-        self.drone_last_ctrl = copy.deepcopy(self.drone.ctrl)
-        self.last_drone_pose = copy.deepcopy(self.drone_pose)
+        self.goal_attributes = (self._calculate_goal_attributes())  # return list of [dx, dy, dz, d]
+        self.has_finished, self.reward = (self._calculate_reward())  # return bool and float
 
         # eg 2.0, True, [0.3, 0.4, 0.1, 5.0], [33.0, 44.0, 55.0], [3.5, 3.5, 3.6, 3.7]
-        return self.reward, self.has_finished, self.goal_attributes, self.drone_rpy, self.drone_motor_thrust
+        return (self.reward, self.has_finished, self.goal_attributes, self.drone_rpy, self.last_drone_motor_thrust, self.drone_motor_thrust, self.last_imu_acc_data, self.imu_acc_data, self.last_imu_gyro_data, self.imu_gyro_data)
```
Comment (PiusLim373) [inline drone_control_gym.py]: why is this return a tuple? does the learning agent need the last motor thrust, last acc and last gyro data?

File: drone_control_gym.py [hunk 5]
```diff
@@ -171,28 +192,25 @@ def step(self, action):
...
 
     def get_last_state(self):
-        return self.reward, self.has_finished, self.goal_attributes, self.drone_rpy, self.drone_motor_thrust
+        return (self.reward, self.has_finished, self.goal_attributes, self.drone_rpy, self.last_drone_motor_thrust, self.drone_motor_thrust, self.last_imu_acc_data, self.imu_acc_data, self.last_imu_gyro_data, self.imu_gyro_data)
```
Comment (PiusLim373) [inline drone_control_gym.py]: why is this return a tuple? does the learning agent need the last motor thrust, last acc and last gyro data?

**Review by PiusLim373 — CHANGES_REQUESTED** [review]

**Review by PiusLim373 — APPROVED** [review]

**Pull request merged**